### PR TITLE
test(all): run pending spec

### LINKED ISSUE #218: Increase spec coverage
Through the use of some fixtures, the system could be essentially end-to-end spec'd in this repo.
This should be achieved by adding `core` as a service to the `docker-compose.yml`.

A place to start with this would be resolving all the `pending` specs.

### PR DESCRIPTION
**Pull request by tassja**
Adds core service to rest-api testing to improve end to end test accuracy
Closes #218

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -15,6 +15,10 @@ x-etcd-client-env: &etcd-client-env
   ETCD_HOST: ${ETCD_HOST:-etcd}
   ETCD_PORT: ${ETCD_PORT:-2379}
 
+x-core-client-env: &core-client-env
+  CORE_HOST: ${CORE_HOST:-core}
+  CORE_PORT: ${CORE_PORT:-3000}
+
```
Comment (caspiano) [inline docker-compose.yml:21]:
```suggestion
```

File: docker-compose.yml [hunk 2]
```diff
@@ -90,3 +95,25 @@ services:
...
+      nofile: 40000
+      core:
+        soft: 0
+        hard: 0
+    environment:
+      # Service Hosts
+      <<: *etcd-client-env
+      <<: *redis-client-env
+      <<: *rethinkdb-client-env
+      <<: *core-client-env
```
Comment (caspiano) [inline docker-compose.yml:117]:
```suggestion
```

File: spec/controllers/drivers_spec.cr [hunk 3]
```diff
@@ -90,34 +95,29 @@ module PlaceOS::Api
...
+          updated.commit.starts_with?("RECOMPILE").should be_false
+          clear_tables
+        end
       end
 
       describe "scopes" do
         before_each do
-          HttpMocks.etcd_range
           HttpMocks.core_compiled
+          # HttpMocks.reset
```
Comment (caspiano) [inline spec/controllers/drivers_spec.cr:120]: I assume we want to enable/disable these mocks depending on whether they depend on core or not.

File: spec/controllers/systems_spec.cr [hunk 4]
```diff
@@ -362,58 +393,58 @@ module PlaceOS::Api
...
+            method: "GET",
+            path: path,
+            headers: authorization_header.merge({"Content-Type" => "application/json"}),
+          )
 
-        puts result.inspect
+          state = Hash(String, String).from_json(response.body)
+          state["nugget"].should eq("1")
+        end
+        clear_tables
```
Comment (caspiano) [inline spec/controllers/systems_spec.cr:447]: What is the explicit `clear_tables` for? Is it to get core into the same default state?

**Review by caspiano — COMMENTED** [review]
This'll be a very useful addition to the spec suites 💯 💯 💯

File: spec/controllers/systems_spec.cr (hunk 4 above)
Comment (tassja) [reply spec/controllers/systems_spec.cr:447]: Yeah

File: spec/controllers/systems_spec.cr [hunk 5]
```diff
@@ -364,58 +395,58 @@ module PlaceOS::Api
         end
       end
 
-      # TODO: awaiting addition of `core` to test environment
-      pending "GET /:sys_id/functions/:module_slug" do
-        cs = Model::Generator.control_system.save!
+      describe "with core" do
```
Comment (caspiano) [inline spec/controllers/systems_spec.cr:398]: For each spec that calls `core`, pls add a `tags: ["core"]` argument to the `it`

That way we can run specs without  core

File: spec/controllers/systems_spec.cr (hunk 4 above)
Comment (caspiano) [reply spec/controllers/systems_spec.cr:449]: To make sure specs run as quick as possible (despite cloning/building drivers), try to get the set up to create or restore to a default state the same set of resources

File: src/placeos-rest-api/controllers/drivers.cr [hunk 6]
```diff
@@ -97,9 +97,10 @@ module PlaceOS::Api
       # Set the repository commit hash to head
       driver.update_fields(commit: "RECOMPILE-#{driver.commit}")
 
-      # Wait until the commit hash is not head with a timeout of 20 seconds
-      Utils::Changefeeds.await_model_change(driver, timeout: 20.seconds) do |update|
-        update.destroyed? || !update.commit.starts_with? "RECOMPILE"
+      # Wait until the commit hash is not head with a timeout of 90 seconds
+      # ameba:disable Style/RedundantReturn
+      return Utils::Changefeeds.await_model_change(driver, timeout: 90.seconds) do |update|
```
Comment (caspiano) [inline src/placeos-rest-api/controllers/drivers.cr:102]: What's the `return` doing?

Comment (tassja) [reply src/placeos-rest-api/controllers/drivers.cr:102]: It doesn't work without the return I tried

Comment (tassja) [reply src/placeos-rest-api/controllers/drivers.cr:102]: Even though it should...

**Review by stakach — APPROVED** [review]
LGTM

**Pull request merged**